external_accounts/giles.py

```diff
@@ -0,0 +1,163 @@
+        try:
+            account = CitesphereAccount.objects.get(user=self.user, repository=self.repository)
+            return account.access_token
+        except CitesphereAccount.DoesNotExist:
+            return None
+        
+    def get_file_content(self, file_id):
+        """
+        Get the content of a file from the Giles API.
+
+        Args:
+            file_id: ID of the file to retrieve content for
+
+        Returns:
+            String containing the file content
+
+        Raises:
+            ValueError: If user is not authenticated with Citesphere
+            HTTPError: If API request fails
+        """
+        if not self.access_token:
+            raise ValueError("User must authenticate with Citesphere before making API calls")
+            
+        headers = {'Authorization': f'Bearer {self.access_token}'}
+        url = f"{self.base_url}/api/v2/resources/files/{file_id}/content/"
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        return response.text
+
+    def check_upload_progress(self, progress_id):
+        """
+        Check the status of an upload using its progress ID
+        
+        Args:
+            progress_id: Progress ID of the upload to check
+            
+        Returns:
+            str|None: The upload ID if available, otherwise None
+        """
+        if not self.access_token:
+            raise ValueError("User must authenticate with Citesphere before making API calls")
+            
+        headers = {'Authorization': f'Bearer {self.access_token}'}
+        url = f"{self.base_url}/api/v2/files/upload/check/{progress_id}"
+        
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        
+        return response.json().get('uploadId')
+
+    def get_upload_details(self, upload_id):
+        """
+        Get details about an upload from Giles API
+        
+        Args:
+            upload_id: ID of the upload to get details for
+            
+        Returns:
+            Dictionary containing upload details including:
+            - Document ID and status
+            - Upload ID and date
+            - Access level
+            - Original uploaded file info
+            - Extracted text file info 
+            - Page images and text
+            
+        Note: If upload is still being processed, returned data may be incomplete
+        """
+        if not self.access_token:
+            raise ValueError("User must authenticate with Citesphere before making API calls")
+            
+        headers = {'Authorization': f'Bearer {self.access_token}'}
+        url = f"{self.base_url}/api/v2/resources/files/upload/{upload_id}"
+        
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        
+        return response.json()[0] # API returns list with single item
+    
+
+def get_giles_document_details(user, file_id, repository):
```

Comment:
let's put those functions into a class that is initialized with user and repository. we'll probably need it more eventually and this way we don't have to keep passing user and repository.